From: Shivananda Hebbar <x0hebbar@ti.com>
Date: Wed, 23 Feb 2011 11:32:02 +0000 (-0600)
Subject: syslink: devh: Fix build break with SYSLINK_RECOVERY disabled
X-Git-Tag: android-2.6.35-2.3-omap4.13.1-Beta~39
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=fc5740817d376cf808f56cfc546b3c7f8058b5ea

syslink: devh: Fix build break with SYSLINK_RECOVERY disabled

If we disable SYSLINK_RECOVERY option in drivers->syslink
->SYSLINK_RECOVERY, build breaks with below errors

drivers/built-in.o: In function `devh_notification_handler':
drivers/dsp/syslink/devh/44xx/devh44xx.c:103: undefined reference to
`ipc_recover_schedule'
drivers/built-in.o: In function `devh44xx_wdt_ipc_notifier_call':
drivers/dsp/syslink/devh/44xx/devh44xx.c:244: undefined reference to
`ipc_recover_schedule'
make: *** [.tmp_vmlinux1] Error 1

Fixed by moving ipc_revover_schedule function under SYSLINK_RECOVERY Flag.

Change-Id: Idb4de38bbe1e2586eebac9f76150616ea93d56f3
Signed-off-by: Shivananda Hebbar <x0hebbar@ti.com>


Index: kernel/arch/arm/plat-omap/include/syslink/ipc.h
===================================================================
--- kernel.orig/arch/arm/plat-omap/include/syslink/ipc.h	2011-06-17 18:25:25.000000000 +0900
+++ kernel/arch/arm/plat-omap/include/syslink/ipc.h	2011-06-19 22:56:02.000000000 +0900
@@ -181,9 +181,8 @@
 /* check if ipc is in recovery state */
 #ifdef CONFIG_SYSLINK_RECOVERY
 bool ipc_recovering(void);
-#endif
 
 /* Indicate to schedule the recovery mechanism */
 void ipc_recover_schedule(void);
-
+#endif /* ifdef CONFIG_SYSLINK_RECOVERY */
 #endif /* ifndef _IPC_H_ */
Index: kernel/drivers/dsp/syslink/devh/44xx/devh44xx.c
===================================================================
--- kernel.orig/drivers/dsp/syslink/devh/44xx/devh44xx.c	2011-06-19 22:23:44.000000000 +0900
+++ kernel/drivers/dsp/syslink/devh/44xx/devh44xx.c	2011-06-19 22:56:02.000000000 +0900
@@ -99,9 +99,10 @@
 	pr_warning("Sys Error occured in Ducati for proc_id = %d\n",
 		proc_id);
 
+#ifdef CONFIG_SYSLINK_RECOVERY
 	/* schedule the recovery */
 	ipc_recover_schedule();
-
+#endif /* ifdef CONFIG_SYSLINK_RECOVERY */
 	devh_notify_event((struct omap_devh *)arg, DEV_SYS_ERROR);
 }
 
@@ -239,9 +240,10 @@
 
 	pr_warning("Ducati Watch Dog fired\n");
 
+#ifdef CONFIG_SYSLINK_RECOVERY
 	/* schedule the recovery */
 	ipc_recover_schedule();
-
+#endif /* ifdef CONFIG_SYSLINK_RECOVERY */
 	while (i--) {
 		obj = devh_get_obj(i);
 		devh_notify_event(obj, DEV_WATCHDOG_ERROR);
